### PR TITLE
Multi clause nodeset

### PR DESCRIPTION
**Pull request by mgeplf**

### CODE REVIEW

File: src/node_sets.cpp [hunk 1]
```diff
@@ -325,118 +353,137 @@ int64_t get_integer_or_throw(const json& el) {
...
 
-NodeSetRules _dispatch_node(const json& contents) {
-    NodeSetRules ret;
+NodeSetRulePtr _dispatch_node(const std::string& attribute, const json& value) {
+    if (value.is_number()) {
+        if (attribute == "population") {
+            throw SonataError("'population' must be a string");
+        }
+
+        int64_t v = get_integer_or_throw(value);
```
Comment (NadirRoGue) [inline src/node_sets.cpp:362]: I guess a function to parse unsigned integers can be added in the future? Not that it will overflow but just a comment.

**Review by NadirRoGue — DISMISSED** [review]
LGTM

File: src/node_sets.cpp (hunk 1 above)
Comment (mgeplf) [reply src/node_sets.cpp:362]: Nope, good call, I will fix that.

**Review by NadirRoGue — APPROVED** [review]
LGTM

**Pull request merged**